backend/src/services/udemy/udemy.service.ts

```diff
@@ -20,20 +28,78 @@ class Udemy {
     this.#httpService = httpService;
   }
 
-  public async getByUrl(url: URL): Promise<UdemyGetResponseDto> {
-    const courseIdOrSlug = url.pathname
-      .split('/')
-      .filter(Boolean)
-      .pop() as string;
+  public async getCourseByUrl(url: URL): Promise<UdemyCourseGetResponseDto> {
+    const { pathname: courseIdOrSlug } = url;
     const headers = this.getHeaders();
-    const res = await this.#httpService.load<UdemyGetResponseDto>(
-      this.getRequestUrl(courseIdOrSlug),
+    const res = await this.#httpService.load<UdemyCourseGetResponseDto>(
+      this.getCourseRequestUrl(courseIdOrSlug),
       { headers, method: HttpMethod.GET },
     );
 
     return res;
   }
 
+  public async getModulesByCourseId(
+    courseId: number,
+  ): Promise<UdemyModuleGetResponseDto[]> {
+    const modules = await this.fetchAllCourseModules(
+      this.#INITIAL_PAGE,
+      courseId,
+    );
+
+    return modules;
+  }
+
+  private async fetchAllCourseModules(
+    page: number,
+    courseId: number,
+  ): Promise<UdemyModuleGetResponseDto[]> {
+    let modules: UdemyModuleGetResponseDto[] = [];
```

Comment:
Let's do it via const and push
```suggestion
    const modules: UdemyModuleGetResponseDto[] = [];
```